### PR TITLE
feat: add `env_vars` as input to `run_sh`

### LINKED ISSUE #2050: Pass env to run_sh
### Background & motivation

It would be great if I could pass env variables to run_sh.

### Desired behaviour

```python
result = plan.run_sh(
        run = "mkdir -p kurtosis && echo $EXAMPLE",
        image = "badouralix/curl-jq",
        files = {
            "/path/to/file/1": files_artifact_1,
            "/path/to/file/2": files_artifact_2,
        },
        env = {
            "EXAMPLE": value
        },
        store = [
            # EXAMPLE: Creates a files artifact named `kurtosis_txt` containing the `kurtosis.txt` file
            StoreSpec(src = "/src/kurtosis.txt", name = "kurtosis_txt"),
            
            # EXAMPLE: Creates a files artifact with an automatically-generated name containing `genesis.json`
            StoreSpec(src = "/genesis.json"),

            # EXAMPLE: Creates a files artifact with an automatically-generated name containing `address.json`
            # This is just syntactic sugar for:
            # StoreSpec(src = "/coinbase/address.json")
            "/coinbase/address.json"
        ],
        wait="180s"
    )
```

### How important is this to you?

Painful; the lack of this feature makes using Kurtosis frictionful.

### What area of the product does this pertain to?

CLI: the Command Line Interface

### PR DESCRIPTION
**Pull request by adschwartz**
## Description:
This change adds `env_vars` to the run_sh. Feature was requested in issue #2050. Example of used in practice:

```
def run(plan, args={}):
    result = plan.run_sh(
        run = "mkdir -p kurtosis && echo $EXAMPLE",
        image = "badouralix/curl-jq",
        env_vars = {
            "EXAMPLE": "this_is_a_test"
        },
    )
```
outputs:

```
Container images used in this run:
> badouralix/curl-jq - locally cached

> run_sh run="mkdir -p kurtosis && echo $EXAMPLE" image="badouralix/curl-jq" env_vars={"EXAMPLE": "this_is_a_test"}
Command returned with exit code '0' and the following output:
--------------------
this_is_a_test

--------------------

Starlark code successfully run. No output was returned.
```

## Is this change user facing?
YES

## References (if applicable):
Closes: #2050

### CODE REVIEW

File: docs/docs/api-reference/starlark-reference/plan.md [hunk 1]
```diff
@@ -477,6 +477,13 @@ The `run_sh` instruction executes a one-time execution task. It runs the bash co
         # OPTIONAL (Default: badouralix/curl-jq)
         image = "badouralix/curl-jq",
 
+        # Defines environment variables that should be set inside the Docker container running the service.
```
Comment (mieubrisse) [inline docs/docs/api-reference/starlark-reference/plan.md]: Think "service" should be "task"

**Review by mieubrisse — APPROVED** [review]

**Review by h4ck3rk3y — APPROVED** [review]

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_sh.go [hunk 2]
```diff
@@ -147,8 +154,21 @@ func (builtin *RunShCapabilities) Interpret(_ string, arguments *builtin_argumen
 		}
 	}
 
+	envVars := map[string]string{}
```
Comment (h4ck3rk3y) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_sh.go]: i wonder whether we can extract this to tasks_shared.go

File: internal_testsuites/golang/testsuite/startosis_run_sh_task_test/run_task_sh_task_test.go [hunk 3]
```diff
@@ -55,6 +55,11 @@ def run(plan):
   result2 = plan.run_sh(run="cat /temp/tech.txt | tr -d '\n'", files={"/temp": file_artifacts[0]})
   plan.verify(result2.output, "==", "kurtosis")
 `
+
+	runShWithEnvVar = `
+def run(plan):
+  result = plan.run_sh(run="mkdir -p kurtosis && echo $EXAMPLE",image="badouralix/curl-jq",env_vars={"EXAMPLE": "value"})
```
Comment (h4ck3rk3y) [inline internal_testsuites/golang/testsuite/startosis_run_sh_task_test/run_task_sh_task_test.go:61]: maybe can add a plan.verify here to test that the value is value

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go [hunk 4]
```diff
@@ -203,8 +203,21 @@ func (builtin *RunPythonCapabilities) Interpret(_ string, arguments *builtin_arg
 		}
 	}
 
+	envVars := map[string]string{}
```
Comment (h4ck3rk3y) [inline core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_python.go]: Need to check if arguments.IsSet(EnvVarsArgname) is set

File: core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_sh.go (hunk 2 above)
Comment (adschwartz) [reply core/server/api_container/server/startosis_engine/kurtosis_instruction/tasks/run_sh.go]: Perfect, thanks for the hint!

File: internal_testsuites/golang/testsuite/startosis_run_sh_task_test/run_task_sh_task_test.go (hunk 3 above)
Comment (adschwartz) [reply internal_testsuites/golang/testsuite/startosis_run_sh_task_test/run_task_sh_task_test.go:61]: Awesome!

**Pull request merged**